connectors/sources/mssql.py

```diff
@@ -75,25 +313,26 @@ def __init__(self, configuration):
             configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
         """
         super().__init__(configuration=configuration)
-
-        # Connector configurations
-        self.retry_count = self.configuration["retry_count"]
-
-        # Connection related configurations
-        self.user = self.configuration["username"]
-        self.password = self.configuration["password"]
-        self.host = self.configuration["host"]
-        self.port = self.configuration["port"]
         self.database = self.configuration["database"]
-        self.tables = self.configuration["tables"]
-        self.engine = None
-        self.connection = None
-        self.certfile = ""
-        self.ssl_enabled = self.configuration["ssl_enabled"]
-        self.ssl_ca = self.configuration["ssl_ca"]
-        self.validate_host = self.configuration["validate_host"]
-        self.queries = MSSQLQueries()
         self.schema = self.configuration["schema"]
+        self.mssql_client = MSSQLClient(
+            host=self.configuration["host"],
+            port=self.configuration["port"],
+            user=self.configuration["username"],
+            password=self.configuration["password"],
+            database=self.configuration["database"],
+            tables=self.configuration["tables"],
+            schema=self.configuration["schema"],
+            ssl_enabled=self.configuration["ssl_enabled"],
+            ssl_ca=self.configuration["ssl_ca"],
+            validate_host=self.configuration["validate_host"],
+            retry_count=self.configuration["retry_count"],
+            fetch_size=self.configuration["fetch_size"],
+            logger_=self._logger,
+        )
+
+    def _set_internal_logger(self):
+        self.mssql_client.set_logger(self._logger)
```

Comment:
Probably not needed since you set it beforehands?

Reply:
When a data source is instantiated, `self._logger` is set to the global `logger`:
https://github.com/elastic/connectors-python/blob/ed88820df1b622c68a5865baba152a47420221bf/connectors/source.py#L347-L349

and set by `set_logger`:
https://github.com/elastic/connectors-python/blob/ed88820df1b622c68a5865baba152a47420221bf/connectors/sync_job_runner.py#L113-L116

The `set_logger` will subsequently call `_set_internal_logger`:

https://github.com/elastic/connectors-python/blob/ed88820df1b622c68a5865baba152a47420221bf/connectors/source.py#L364-L366

We do this because we don't want to change the signature of `BaseDataSource`
https://github.com/elastic/connectors-python/blob/ed88820df1b622c68a5865baba152a47420221bf/connectors/source.py#L347